main.go

```diff
@@ -130,7 +130,13 @@ func Load(filename string, offset int, overlay map[string][]byte) (*packages.Pac
 			mode |= parser.Trace
 		}
 		file, err := parser.ParseFile(fset, fname, src, mode)
-		if file == nil || err != nil {
+		if file == nil {
+			if err==nil {
```

Comment:
`ParseFile` will never return a nil file and a nil error, so this `if err == nil` check can be removed and the outer if can be reverted back to the original `if file == nil || err != nil`.

Reply:
no it can not reverted back to `if file == nil || err != nil`
i want to get documentation even if my file contains errors, if you revert it, you only get documentation if your file does not contain any errors

Reply:
Oh, good call.  I maintain that both `file` and `err` won't both be nil, so how about this?

```
if file == nil {
  if isInputFile {
    ch <- result{nil, err}
  }
  return nil, err
}
```

Reply:
yeah, i started out with that, and then i thought, what if  for some strange reason err is nil???
and then i added the the `if err==nil` part, just to be save ....